lib/active_merchant/billing/gateways/sum_up.rb

```diff
@@ -157,7 +162,7 @@ def parse(body)
       end
 
       def success_from(response)
-        return true if response == 204
+        return true if (response.is_a?(Hash) && response[:next_step]) || response == 204
```

Comment:
❓  Question:
why it's required to validate if the response is a Hash? what are the other possible values?

Reply:
Yes, this is because the refund transaction return an Integer, If we don't verify the Hash then it breaks when doing a refund

Reply:
Could this possibly lead to false positives? Would the presence of a `Hash` always mean a successful txn or are there failed transactions that return a hash with error messages?